src/frontend/src/handler/alter_table_column.rs

```diff
@@ -74,7 +74,11 @@ pub async fn handle_alter_table_column(
 
     if let Some(source_schema) = &source_schema {
         if schema_has_schema_registry(source_schema) {
-            bail_not_implemented!("Alter table with source having schema registry");
+            return Err(ErrorCode::NotSupported(
+                "alter table with schema registry".to_string(),
+                "try `ALTER TABLE .. FORMAT .. ENCODE .. (...)` instead".to_string(),
```

Comment:
In this case, I guess the user may actually want to refresh the schema i.e. pulling from the schema registry to get a latest schema. Do we have such syntax support?

Reply:
>Do we have such syntax support?

Yes, that's exactly what this pr has done: impl `ALTER TABLE .. FORMAT .. ENCODE .. (schema.xxx = ...)`.

Reply:
I mean, he might not want to change the `schema.location` or `message` but only do a refresh from the schema registry (because he has updated the schema in the schema registry server before)

Reply:
Got it. In current impl, users need to repeat the same config as they create the table/source to do a refresh operation. Maybe we can add a shortcut for this purpose:
1. `ALTER ... FORMAT .. ENCODE ..;`: empty `format_encode_options` implies a simple refresh.
2. `ALTER ... REFRESH SCHEMA REGISTRY;`: new syntax to avoid confusion in original `FORMAT ENCODE` syntax.

Reply:
Is there a plan to allow users to specify a completely different set of `FORMAT` and `ENCODE` when altering a table in the future? If not, I would suggest using a syntax like `ALTER ... REFRESH SCHEMA REGISTRY` for clarity and convenience.

Reply:
What does `ALTER ... REFRESH SCHEMA REGISTRY` actually mean? If we support one day we need to support independent `SCHEMA REGISTRY` for key/value or even kafka header. We need extra work to design semantic for the new statment.... So -1 for this  syntax

Reply:
>I'd suggest to alter the connector part to NULL and then add a new one in this case

Can you provide an example? Not quite get it.

Reply:
> What does ALTER ... REFRESH SCHEMA REGISTRY actually mean? If we support one day we need to support independent SCHEMA REGISTRY for key/value or even kafka header. We need extra work to design semantic for the new statment.... So -1 for this syntax

You can simply refresh both key & value's schema. If a schema is not updated in schema registry side, the command won't make any changes.

This is a shortcut if users updated the schema in schema registry side, but didn't change anything else. Letting them specifying all fields (schema registry endpoint, message name, etc.) will scare them.

Reply:
🤔 I think they can get all the fields easily from the system table. 
Both LGTM.

Reply:
Do we have a decision here?

FYI, we are going to implement `alter source src format xxx encode xxx (connector = 'new_connector')`. As users may have a source for history data and another for incremental data, they can continue the consumption by altering the connector.

So we gonna mix `schema refresh` and `connector reset` in the same sql, and probably more cases in the future. The semantic can be confusing then. I suggest we designate the sql individually.

Reply:
`ALTER ... REFRESH SCHEMA`

Reply:
`ALTER SOURCE ... FORMAT .. ENCODE ..`

Reply:
Actually I believe these two features could be clearly distinguished. 🤔


Refresh schema (or update any other minor configuration fields)
- The connector is still the same.
- The state of the source executors is kept.
- Users would not like to **repeat** unchanged properties.

Alter connector
- The connector is replaced with a completely different one.
- The state of the source executors is reset.
- Users would not like to **reuse** any existing properties.

When it comes to designing syntax, I suppose we should choose different syntaxes based on the specific requirements. For example, we may leave the general syntax of `ALTER SOURCE ... FORMAT .. ENCODE ..` for the latter one. For the former feature, a sugar like `REFRESH SCHEMA` looks good to me if we don't find any other similar requirements.